tests/integration/test_vulnerability_detector/test_providers/data/test_cases/test_no_os.yaml

```diff
@@ -0,0 +1,59 @@
+-
+  name: 'Amazon Linux'
+  description: 'Amazon Linux provider'
+  configuration_parameters:
+    PROVIDER: 'alas'
+  metadata:
+    provider_name: 'Amazon Linux'
+
+-
+  name: 'Canonical'
+  description: 'Canonical provider'
+  configuration_parameters:
+    PROVIDER: 'canonical'
+  metadata:
+    provider_name: 'Ubuntu'
+
+-
+  name: 'Debian'
+  description: 'Debian provider'
+  configuration_parameters:
+    PROVIDER: 'debian'
+  metadata:
+    provider_name: 'Debian'
+
```

Comment:
There are several places where you do not define the variable `os` although in others you do even though they are empty. It does not have a fixed structure, which can cause `KeyError` in the test.

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/2547/commits/90bed077f7ebf72dd21e5d15b1af1bc674ba3704